tools/internal_ci/linux/grpc_xds_k8s.sh

```diff
@@ -45,6 +45,9 @@ build_test_app_docker_images() {
   gcloud -q auth configure-docker
   docker push "${CLIENT_IMAGE_NAME}:${GIT_COMMIT}"
   docker push "${SERVER_IMAGE_NAME}:${GIT_COMMIT}"
+  branch_name=$(echo "$KOKORO_JOB_NAME" | sed -E 's|^grpc/core/(.+)/linux/grpc_xds_k8s$|\1|')
```

Comment:
I think the regexp won't work with branches. The pattern of jobs from master and jobs from branches is different:

1. master: `grpc/core/master/linux/grpc_xds_k8s`
2. branches: `grpc/core/<branch_name>/branch/linux/grpc_xds_k8s`

Reply:
Good catch! Changed to matching prefix only:

```
branch_name=$(echo "$KOKORO_JOB_NAME" | sed -E 's|^grpc/core/(.+)/|\1|')
```